src/main/java/org/springframework/data/elasticsearch/core/AbstractElasticsearchTemplate.java

```diff
@@ -455,11 +465,39 @@ protected void maybeCallbackBeforeConvertWithQuery(Object query) {
 	}
 
 	// this can be called with either a List<IndexQuery> or a List<UpdateQuery>; these query classes
-	// don't have a common bas class, therefore the List<?> argument
+	// don't have a common base class, therefore the List<?> argument
 	protected void maybeCallbackBeforeConvertWithQueries(List<?> queries) {
 		queries.forEach(this::maybeCallbackBeforeConvertWithQuery);
 	}
 
+	protected <T> T maybeCallbackAfterSave(T entity) {
+
+		if (entityCallbacks != null) {
+			return entityCallbacks.callback(AfterSaveCallback.class, entity);
```

Comment:
We should make sure that the value returned from the callback is not `null`. Although the API has neither the parameter or return value marked as nullable, we might get a null value by some bad implemention of an after-save callback. So I think we should assert this here.

Reply:
Looking at the code of `SimpleEntityCallbackInvoker`, it looke like it makes sure that the value returned by a callback is non-null:

```
				throw new IllegalArgumentException(
						String.format("Callback invocation on %s returned null value for %s", callback.getClass(), entity));
```
Also, javadoc for `EntityCallbackInvoker#invokeCallback()` says that the returned value is 'never null'.
So even a callback that returns null will just cause an exception somewhere in EntityCallbacks implementation, and here we will not get a null value.

Reply:
ah, I didn't dig this deep into the code for this PR, sorry (wish this were Kotlin stuff!), so that's ok